keps/sig-api-machinery/20190325-unions.md

```diff
@@ -214,24 +238,74 @@ below.
+| A15      | `discriminator: B`<br>`b: {...}` | `discriminator: A`<br>`a: {...}` | `discriminator: B`<br>`b: {...}` |
+
+Example updates with invalid content in the new object:
+
+| Scenario | New object                       | Existing object                  | Result                                                                      |
+|----------|----------------------------------|----------------------------------|-----------------------------------------------------------------------------|
+| B1       | `a: {...}, b: {...}`             | empty                            | fail<br>(two new fields set)                                                |
+| B2       | `a: {...}, b: {...}`             | `a: {...}`                       | fail<br>(two fields set, no discriminator)                                  |
+| B3       | `a: {...}, b: {...}`             | `discriminator: A`<br>`a: {...}` | fail<br>(two fields set, no discriminator)                                  |
+| B4       | `a: {...}, b: {...}`             | `discriminator: B`<br>`b: {...}` | fail<br>(two fields set, no discriminator)                                  |
+| B5       | `discriminator: A`<br>`b: {...}` | empty                            | fail<br>(new field and changed discriminator don't match)                   |
+| B6       | `discriminator: A`<br>`b: {...}` | `a: {...}`                       | fail<br>(new field and changed discriminator don't match)                   |
+| B8       | `discriminator: B`<br>`a: {...}` | empty                            | fail<br>(new field and changed discriminator don't match)                   |
+| B7       | `discriminator: A`<br>`b: {...}` | `discriminator: A`<br>`a: {...}` | fail<br>(discriminator doesn't match field set)                             |
+| B9       | `discriminator: B`<br>`a: {...}` | `a: {...}`                       | `discriminator: B`<br>(discriminator changed)                               |
+| B10      | `discriminator: B`<br>`a: {...}` | `discriminator: A`<br>`a: {...}` | `discriminator: B`<br>(discriminator changed)                               |
+
+If the discriminator is "deducible", then B7 becomes:
+| B7       | `discriminator: A`<br>`b: {...}` | `discriminator: A`<br>`a: {...}` | `discriminator: B`<br>`b:{...}`<br>(field replace, discriminator unchanged) |
+
+### Normalizing on server-side apply
+
+Normalizing on server-side apply is much simpler, since the user sends their
+exact intent. It's mostly easier to validate that the intent is correct. It's
+important to note that implicit discriminator changes (for example if you change
+a field without changing the discriminitaor) are owned by the user, which might
+result in conflicts.
+
+The process works as follow:
+- If more than one field is set, fail,
+- If the discriminator doesn't match the field set, fail,
+- If the discriminator isn't set, set it if possible.
```

Comment:
only if deducible, right? we don't want to enable skipping setting required discriminators for new types, right?

Reply:
Right, hopefully it would be required for new types.